sim/dex.ts

```diff
@@ -518,6 +518,16 @@ export class ModdedDex {
 			for (const dataType of DATA_TYPES) {
 				const parentTypedData: DexTable<any> = parentDex.data[dataType];
 				const childTypedData: DexTable<any> = dataCache[dataType] || (dataCache[dataType] = {});
+				// if child is empty and there's no Scripts.init, there's no need to copy, just re-use.
+				let childIsEmpty = true;
+				for (const k in childTypedData) { // eslint-disable-line @typescript-eslint/no-unused-vars
+					childIsEmpty = false;
+					break;
+				}
```

Comment:
There's a world in which `loadDataFile` would return `AnyObject | void` but this isn't it. (Well, not yet, at least.)

Reply:
Do you mean you want it not to throw exceptions? That doesn't make much sense. You would want to refuse starting up, and even halting hotpatches if the data files have syntax errors or whatever.

Reply:
No, I think it shouldn't return a new empty object if the file doesn't exist. In the case of `Scripts`, the caller can do this, while in the case of the data files, this check would simplify into a boolean test.

Reply:
That indeed sounds like an alternate universe.